browser/ui/startup/brave_obsolete_system_infobar_delegate.cc

```diff
@@ -28,19 +34,66 @@ BraveObsoleteSystemInfoBarDelegate::BraveObsoleteSystemInfoBarDelegate() =
 BraveObsoleteSystemInfoBarDelegate::~BraveObsoleteSystemInfoBarDelegate() =
     default;
 
-int BraveObsoleteSystemInfoBarDelegate::GetButtons() const {
-  return BUTTON_OK;
+bool BraveObsoleteSystemInfoBarDelegate::HasCheckbox() const {
+  return true;
 }
 
-std::u16string BraveObsoleteSystemInfoBarDelegate::GetButtonLabel(
-    InfoBarButton button) const {
+std::u16string BraveObsoleteSystemInfoBarDelegate::GetCheckboxText() const {
   return l10n_util::GetStringUTF16(
       IDS_OBSOLERE_SYSTEM_INFOBAR_DONT_SHOW_BUTTON);
 }
 
-bool BraveObsoleteSystemInfoBarDelegate::Accept() {
+void BraveObsoleteSystemInfoBarDelegate::SetCheckboxChecked(bool checked) {
+  launch_confirmation_dialog_ = checked;
+}
+
+bool BraveObsoleteSystemInfoBarDelegate::InterceptClosing() {
+  if (!launch_confirmation_dialog_)
+    return false;
+
+  // This infobar will be destroyed after confirmation dialog closed.
+  brave::ShowObsoleteSystemConfirmDialog(base::BindOnce(
+      &BraveObsoleteSystemInfoBarDelegate::OnConfirmDialogClosing,
+      weak_factory_.GetWeakPtr()));
+  return true;
+}
+
+int BraveObsoleteSystemInfoBarDelegate::GetButtons() const {
+  return BUTTON_NONE;
+}
+
+void BraveObsoleteSystemInfoBarDelegate::OnConfirmDialogClosing(bool suppress) {
+  infobar()->RemoveSelf();
+
```

Comment:
nit: Can we add a comment that "this" object is destroyed at this point so that we can take care about it.